### PR TITLE
[ci] Add Ubuntu 24.04 and Fedora40 builds

### PR DESCRIPTION
**Pull request by dpiparo**

### CODE REVIEW

**Comment by dpiparo** [comment]
A few failing tests so far:
```
2:pyunittests-distrdf-unit-test-friendinfo
3:pyunittests-distrdf-unit-test-headnode
8:pyunittests-distrdf-unit-test-ranges
10:pyunittests-distrdf-unit-backend-test-dist
11:pyunittests-distrdf-unit-backend-test-graph-caching
1192:tutorial-dataframe-df007_snapshot-py
1385:tutorial-tmva-TMVA_CNN_Classification-py
1386:tutorial-tmva-TMVA_Higgs_Classification-py
1670:roottest-python-distrdf-backends-test_all
```
@lmoneta @vepadulano

**Review by guitargeek — APPROVED** [review]
Thank you for this PR!

I would not make merging this PR conditional on fixing the tests. We don't have any platform with Python 3.12 in the CI yet, so adding Ubuntu 24.04 is very important especially for PyROOT development.

Can we maybe temporarily set `tmva=OFF` and `dataframe=OFF` and merge the PR? Or is there maybe another solution? 


*Edit:* sorry I was wrong, we do have Python 3.12 in the CI with Fedora 39. I checked the Python versions on a few platforms.

```txt
mac13          3. 9.6
mac14          3. 9.6
mac-beta       3. 9.6
Windows 10 x64 3.11.7
Windows 10 x86 3.11.7
fedora39       3.12.2
ubuntu2310     3.11.6
ubuntu2404     3.12.3
```

Then `ubuntu24` is not so urgent from that point of view after all.

File: .github/workflows/root-ci.yml [hunk 1]
```diff
@@ -356,6 +356,8 @@ jobs:
             overrides: ["imt=Off", "LLVM_ENABLE_ASSERTIONS=On", "CMAKE_BUILD_TYPE=Debug"]
           - image: ubuntu2310
             overrides: ["LLVM_ENABLE_ASSERTIONS=On", "CMAKE_CXX_STANDARD=20"]
+          - image: ubuntu2404
```
Comment (guitargeek) [inline .github/workflows/root-ci.yml:359]:
```suggestion
          - image: ubuntu24
```
For the sake of consistency with previous LTS releases in the CI: maybe drop the month and just call it `ubuntu24`? Or was this deliberate?

**Comment by guitargeek** [comment]
A suggestion to maybe fix the TMVA failures: maybe It's the openblas incompatibility with NumPy that we also saw on alma9, which we can about by disabling `tmva-cpu`?

 * https://github.com/root-project/root/commit/47c00beb9f25c9c60a8b4a2b74ff9f70d187e10c
 * https://github.com/root-project/root/issues/15309

File: .github/workflows/root-ci.yml (hunk 1 above)
Comment (dpiparo) [reply .github/workflows/root-ci.yml:359]: It was deliberate to be clear about the exact version.

**Comment by dpiparo** [comment]
I disabled tmva-cpu

File: .github/workflows/root-ci-config/buildconfig/fedora40.txt [hunk 2]
```diff
@@ -0,0 +1,4 @@
+builtin_nlohmannjson=On
+builtin_vdt=On
+pythia8=Off
+tmva-pymva=Off
```
Comment (guitargeek) [inline .github/workflows/root-ci-config/buildconfig/fedora40.txt:4]:
```suggestion
```

As of #15439, this is not done done anymore for Fedora 39. Probably for Fedora 40, we shouldn't do this customization either

**Comment by guitargeek** [comment]
About Fedora 40 compression size change, I recently merged this PR that already changed the thresholds:
https://github.com/root-project/root/pull/14295

I understand now from the last group meeting that it would be better to not change these thresholds, because we want to ensure small compression size.

So maybe, that PR needs to be reverted?

**Comment by ferdymercury** [comment]
Related: https://github.com/root-project/root/issues/15621

**Comment by guitargeek** [comment]
Ubuntu 24.04 and Fedora40 are now both in the CI already.

**Pull request closed without merging**